imap_processing/ialirt/l0/process_hit.py

```diff
@@ -0,0 +1,210 @@
+    start_sc_tick = start_sc_ticks.data.min()
+    last_sc_ticks = data["hit_sc_tick"][([data["hit_subcom"] == subcom_range[-1]][-1])]
+    last_sc_tick = last_sc_ticks.data.max()
+
+    grouped_data = data.where(
+        (data["hit_sc_tick"] >= start_sc_tick) & (data["hit_sc_tick"] <= last_sc_tick),
+        drop=True,
+    )
+
+    # Create group labels
+    group_labels = np.searchsorted(
+        start_sc_ticks, grouped_data["hit_sc_tick"], side="right"
+    )
+    grouped_data["group"] = ("group", group_labels)
+
+    return grouped_data
+
+
+def create_l1(
+    fast_rate_1: xr.DataArray,
+    fast_rate_2: xr.DataArray,
+    slow_rate: xr.DataArray,
+) -> dict[str, float]:
+    """
+    Create L1 data dictionary.
+
+    Parameters
+    ----------
+    fast_rate_1 : xr.DataArray
+        Fast rate 1 DataArray.
+    fast_rate_2 : xr.DataArray
+        Fast rate 2 DataArray.
+    slow_rate : xr.DataArray
+        Slow rate DataArray.
+
+    Returns
+    -------
+    l1 : dict
+        Dictionary containing parsed L0 packet data.
+    """
+    fast_rate_1_dict = {
+        prefix: value
+        for prefix, value in zip(HITPrefixes.FAST_RATE_1.value, fast_rate_1.data)
+    }
+    fast_rate_2_dict = {
+        prefix: value
+        for prefix, value in zip(HITPrefixes.FAST_RATE_2.value, fast_rate_2.data)
+    }
+    slow_rate_dict = {
+        prefix: value
+        for prefix, value in zip(HITPrefixes.SLOW_RATE.value, slow_rate.data)
+    }
+
+    l1 = {**fast_rate_1_dict, **fast_rate_2_dict, **slow_rate_dict}
+
+    return l1
+
+
+def process_hit(xarray_data: xr.Dataset) -> list[dict]:
+    """
+    Create L1 data dictionary.
+
+    Parameters
+    ----------
+    xarray_data : dict(xr.Dataset)
+        Dictionary of xarray data including a single
+        set for processing.
+
+    Returns
+    -------
+    hit_data : dict
+        Dictionary final data product.
+    """
+    hit_data = []
+    grouped_data = find_groups(xarray_data)
+
+    for group in np.unique(grouped_data["group"]):
+        fast_rate_1 = grouped_data["hit_fast_rate_1"][
+            (grouped_data["group"] == group).values
+        ]
+        fast_rate_2 = grouped_data["hit_fast_rate_2"][
+            (grouped_data["group"] == group).values
+        ]
+        slow_rate = grouped_data["hit_slow_rate"][
+            (grouped_data["group"] == group).values
+        ]
+        met = int(grouped_data["hit_met"][(grouped_data["group"] == group).values][0])
+
+        # Verify that each group has 60 datapoints
+        if len(slow_rate.data) != 60:
+            logger.info(
+                f"Incorrect number of packets: {len(slow_rate.data)} "
+                f"for met {met}. Skipping."
+            )
+            continue
+
+        l1 = create_l1(fast_rate_1, fast_rate_2, slow_rate)
+
+        hit_data.append(
+            {
+                "met": met,
+                "hit_lo_energy_e_A_side": float(
```

Comment:
Why are we casting to Python ints/floats? The dataarrays should be able to be added together I think.